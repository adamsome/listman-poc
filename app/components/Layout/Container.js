import React from 'react'

const Container = ({ children, centered }) => {
  let className = "container"
  if (centered) className += " has-text-centered"
  return (
    <div className={className}>
      {children}
    </div>
  )
}

export default Container
