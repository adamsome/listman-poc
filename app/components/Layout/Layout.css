@media screen and (min-width: 980px) {
  .layout {
    margin: 0 auto;
    max-width: 960px;
  }
}

@media screen and (min-width: 1180px) {
  .layout {
    max-width: 1200px;
  }
}

.layout {
  position: relative;
  width: 100%;
}

.layout__nav {
  padding: 0 2.25%;
  height: 3em;
}

.layout__nav-header {
  margin-right: 10px;
  padding-right: 10px;
}

.layout__nav-link {
  margin: 0 16px;
  line-height: 2em;
}

.layout__content {
  min-height: 90vh;
  padding: 0 2.25%;
}

.layout__footer {
  padding: 0 2.25%;
  height: 2em;
  line-height: 2em;
}
