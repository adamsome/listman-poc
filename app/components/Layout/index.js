import React from 'react'

const Layout = ({ children }) => {
  return (
    <div>
      <section className="section">
        {children}
      </section>
      <footer className="footer">
        <div className="container has-text-centered">
          v0.1: No Functionality (
          <a href="http://github.com/adamsome/listman">
            github.com/adamsome/listman
          </a>
          )
        </div>
      </footer>
    </div>
  )
}

export default Layout
