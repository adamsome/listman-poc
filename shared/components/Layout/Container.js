import React from 'react'

const Container = ({ children, className, centered, spaceAbove }) => {
  let fullClass = className + " container"
  if (centered) fullClass += " has-text-centered"

  let style = {}
  if (spaceAbove) style.marginTop = '2rem'

  return (
    <div className={fullClass} style={style}>
      {children}
    </div>
  )
}

export default Container
