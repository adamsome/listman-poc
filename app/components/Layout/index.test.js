import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Layout from './index'

let wrapper
beforeEach(() => {
  wrapper = shallow(<Layout />)
})

it('renders correctly', () => {
  expect(toJson(wrapper)).toMatchSnapshot()
})

