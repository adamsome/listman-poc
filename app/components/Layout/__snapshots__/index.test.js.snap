exports[`test renders correctly 1`] = `
<div>
  <section
    className="section" />
  <footer
    className="footer">
    <div
      className="container has-text-centered">
      v0.1: No Functionality (
      <a
        href="http://github.com/adamsome/listman">
        github.com/adamsome/listman
      </a>
      )
    </div>
  </footer>
</div>
`;
